client/app/pages/queries/components/QueryVisualizationTabs.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { find, orderBy } from 'lodash';
+import Tabs from 'antd/lib/tabs';
+import { VisualizationRenderer } from '@/visualizations/VisualizationRenderer';
+import Button from 'antd/lib/button';
+
+const { TabPane } = Tabs;
+
+export default function QueryVisualizationTabs({ visualizations, queryResult, currentVisualizationId }) {
+  const tabsProps = {};
+  if (find(visualizations, { id: +currentVisualizationId })) {
+    tabsProps.activeKey = `${currentVisualizationId}`;
+  }
+
+  const orderedVisualizations = orderBy(visualizations, ['id']);
+
+  return (
+    <Tabs {...tabsProps} tabBarExtraContent={(<Button><i className="fa fa-plus m-r-5" />New Visualization</Button>)}>
+      {orderedVisualizations.map(visualization => (
+        <TabPane key={`${visualization.id}`} tab={visualization.name}>
+          <VisualizationRenderer
+            visualization={visualization}
+            queryResult={queryResult}
+            context="query"
+          />
+        </TabPane>
+      ))}
+    </Tabs>
+  );
+}
+
+QueryVisualizationTabs.propTypes = {
+  queryResult: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+  visualizations: PropTypes.arrayOf(PropTypes.object),
+  currentVisualizationId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
```

Comment:
How could this prop be a string? Maybe, let's restrict it to number and cast type when passing props to this component?

Reply:
The idea was to accept string id's and cast them in the component, but yeah, this can be done outside, so it's more consistent.